src/librustc/mir/mod.rs

```diff
@@ -2760,18 +2806,36 @@ impl<'tcx> TypeFoldable<'tcx> for Terminator<'tcx> {
 
 impl<'tcx> TypeFoldable<'tcx> for Place<'tcx> {
     fn super_fold_with<'gcx: 'tcx, F: TypeFolder<'gcx, 'tcx>>(&self, folder: &mut F) -> Self {
-        match self {
-            &Place::Projection(ref p) => Place::Projection(p.fold_with(folder)),
-            _ => self.clone(),
+        Place {
+            base: self.base.fold_with(folder),
+            elems: self.elems.fold_with(folder),
```

Comment:
This... just works? Does it know how to reintern itself?

Reply:
`TypeFoldable` are already impl'd for `PlaceBase` and `ProjectonElem(PlaceElem aliases on it)` there, so I "stolen" the way this trait is impl'd for other slice [like](https://doc.rust-lang.org/nightly/nightly-rustc/src/rustc/ty/structural_impls.rs.html#757-766) `&'tcx Slice<Ty<'tcx>>` to `impl TypeFoldable for Slice<PlaceElem>`.
now both `PlaceBase` and `Slice<PlaceElem>` are `TypeFoldable`, so they can intern themselves, am I right?